spec/downloadManager.spec.ts

```diff
@@ -0,0 +1,122 @@
+        };
+
+        shallow(React.createElement(DownloadManager));
+
+        ipcRenderer.send('downloadCompleted',objectDownloadCompleted);
+
+        expect(spy).toBeCalledWith({
+            items: [{
+                _id: 1,
+                fileName: 'test.png',
+                savedPath: 'path://test',
+                total: 1,
+                flashing: false,
+            }], showMainComponent: true,
+        });
+    });
+
+    describe('download element ready', () => {
+        let wrapper: ShallowWrapper;
+        beforeEach(() => {
+            wrapper = shallow(React.createElement(DownloadManager));
+
+            wrapper.setState({
+                items: [
+                    {
+                        _id: 1,
+                        fileName: 'test.png',
+                        savedPath: 'path://test',
+                        total: 1,
+                        flashing: false,
+                    }],
+            });
+        });
+        it('should exist `download-element` class when there are items', () => {
+            expect(wrapper.find('.download-element')).toHaveLength(1);
+        });
+
+        it('should be called `onOpenFile` correctly', () => {
+            const { ipcRenderer } = require('./__mocks__/electron');
+            const sendEventLabel: string = 'send';
+            const spy: jest.SpyInstance = jest.spyOn(ipcRenderer, sendEventLabel);
+
+            wrapper.find('#download-open').simulate('click');
+
+            expect(spy).toBeCalledWith('symphony-api', {
+                cmd: 'download-manager-action',
+                path: 'path://test',
+                type: 'open',
+             });
+        });
+
+        it('should be called `showInFinder` correctly', () => {
+            const { ipcRenderer } = require('./__mocks__/electron');
+            const sendEventLabel: string = 'send';
+            const spy: jest.SpyInstance = jest.spyOn(ipcRenderer, sendEventLabel);
+
+            wrapper.find('#download-show-in-folder').simulate('click');
+
+            expect(spy).toBeCalledWith('symphony-api', {
+                cmd: 'download-manager-action',
+                path: 'path://test',
+                type: 'show',
+             });
+        });
+    });
+
+    describe('download completed event', () => {
+        const { ipcRenderer } = require('./__mocks__/electron');
+        const downloadCompletedLabelEvent: string = 'downloadCompleted';
+        const onLabelEvent: string = 'on';
+        const removeListenerLabelEvent: string = 'removeListener';
+
+        it('should call the `downloadCompleted event when mount component', () => {
+            const spy: jest.SpyInstance = jest.spyOn(ipcRenderer, onLabelEvent);
+
+            shallow(React.createElement(DownloadManager));
+
+            expect(spy).toBeCalledWith(downloadCompletedLabelEvent, expect.any(Function));
+        });
+
+        it('should remove listen `downloadCompleted` when unmount component', () => {
```

Comment:
please change the name to `should remove listen downloadCompleted when component is unmounted`